### PR TITLE
RDoc-2464_LongCountAsync

### PR DESCRIPTION
**Pull request by reebhub**
Edit querying statistics page, Add a page dedicated to counting methods (Count/CountAsync, LongCount/LongCountAsync)

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown [hunk 1]
```diff
@@ -0,0 +1,54 @@
+﻿# Count Query Results  
+
+---
+
+{NOTE: }
+
+* Count query results using one of the methods described below.  
+
+* In This Page:  
+    * [CountAsync](../../../client-api/session/querying/how-to-count-query-results#countasync)
```
Comment (shiranshalom) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown:10]: Can we add here also `Count` and specify that:

- `Count` (`System.Linq`)  
- `CountAsync` (our linq extension & `System.Linq`) - will throw if number is greater than `int.max` value
- `LongCount` (our linq extension & `System.Linq`) - readers should be aware they have these two alternatives and choose accordingly
- `LongCountAsync` (our linq extension)

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    * [CountAsync](../../../client-api/session/querying/how-to-count-query-results#countasync)  
+    * [LongCount and LongCountAsync](../../../client-api/session/querying/how-to-count-query-results#longcount-and-longcountasync)  
+
+{NOTE/}
+
+---
+
+{PANEL: CountAsync}
+
+To count query results where an `Int32` variable is sufficient, use the `CountAsync` method.
```
Comment (shiranshalom) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown:19]: Maybe it is worth mentioning this also for using info from the query stats - the readers should use either `TotalResults` or `LongTotalResults` :-)

Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown:19]: yeah i also updated `client-api/session/querying/how-to-get-query-statistics` and interlinked the two pages.

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/querying/how-to-count-query-results.markdown:10]: done

**Review by shiranshalom — APPROVED** [review]

**Pull request merged**